lib/rmt/file_utils.rb

```diff
@@ -0,0 +1,18 @@
+require 'fileutils'
+
+class RMT::FileUtils
+
+  def self.deduplicate(checksum_type, checksum_value, destination)
+    local_path = DownloadedFile.get_local_path_by_checksum(checksum_type, checksum_value)
+    return false unless local_path
+
+    if RMT::Config.deduplication_by_hardlink?
+      ::FileUtils.ln(local_path, destination)
```

Comment:
If the FS doesn't support hardlinks, this would die with:
```
.rvm/rubies/ruby-2.4.1/lib64/ruby/2.4.0/fileutils.rb:287:in `link': Operation not permitted @ rb_file_s_link - (rmt/public/repo/SUSE/Updates/SLE-SERVER-INSTALLER/12-SP2/x86_64/update/x86_64/hwinfo-21.23_21.38-1.13_15.7.2.x86_64.drpm, rmt/public/repo/SUSE/Updates/SLE-DESKTOP-INSTALLER/12-SP2/x86_64/update/x86_64/hwinfo-21.23_21.38-1.13_15.7.2.x86_64.drpm) (Errno::EPERM)
	from .rvm/rubies/ruby-2.4.1/lib64/ruby/2.4.0/fileutils.rb:287:in `block in ln'
	from .rvm/rubies/ruby-2.4.1/lib64/ruby/2.4.0/fileutils.rb:1476:in `fu_each_src_dest0'
```
It's actually OK that it dies, but it would be nice to have a better error message that says something along the lines of `Your filesystem doesn't allow hardlinks, try using copy instead`.